<launch>
  <arg name="gui" default="true" />
  <arg name="sample_rate" default="44100"/>
  <arg name="abb" default="false" />
  <arg name="kuka" default="false" />
  <arg name="rviz" default="true"/>
  <arg name="thermal" default="false"/>
  <arg name="vision" default="true"/>
  <arg name="pylon" default="true"/>
  <arg name="acoustic" default="true"/>
  <arg name="robot_launch" default="false" />
  <arg name="image_processing" default="true"/>
  <arg name="image_view" default="false" />
  <arg name="defect_prediction" default="true" />
  <arg name="plotjuggler_viz" default="false"/>
  <arg name="plotjuggler_viz_multimodal" default="false"/>
  <arg name="audio_topic" default="/audio"/>
  <arg name="audio_stamped_topic" default="/audioStamped"/>
  <arg name="device" default="hw:2,0" />   <!-- arecord -l -->
  <arg name="video_device" default="/dev/video0" />
  <arg name="gui_camera" default="false" />
  <arg name="frame_rate" default="60.0" />
  <arg name="image_topic_raw" default="/pylon_camera_node/image_raw"/> <!-- "/usb_cam/image_raw" or "/pylon_camera_node/image_raw" -->


  <include file="$(find multimodal_monitoring)/launch/multimodal_monitoring.launch">
    <arg name="sample_rate" value="$(arg sample_rate)" />
    <arg name="abb" value="$(arg abb)" />
    <arg name="kuka" value="$(arg kuka)" />
    <arg name="rviz" value="$(arg rviz)" />
    <arg name="thermal" value="$(arg thermal)" />
    <arg name="vision" value="$(arg vision)" />
    <arg name="pylon" value="$(arg pylon)" />
    <arg name="acoustic" value="$(arg acoustic)" />
    <arg name="robot_launch" value="$(arg robot_launch)" />
    <arg name="image_processing" value="$(arg image_processing)" />
    <arg name="image_view" value="$(arg image_view)"  />
    <arg name="defect_prediction" value="$(arg defect_prediction)" />
    <arg name="plotjuggler_viz" value="$(arg plotjuggler_viz)" />
    <arg name="plotjuggler_viz_multimodal" value="$(arg plotjuggler_viz_multimodal)" />
    <arg name="audio_topic" value="$(arg audio_topic)" />
    <arg name="device" value="$(arg device)" />
    <arg name="gui_camera" value="$(arg gui_camera)" />
    <arg name="video_device" value="$(arg video_device)" />
    <arg name="image_topic_raw" value="$(arg image_topic_raw)" />
    <arg name="frame_rate" value="$(arg frame_rate)" />
  </include>

  <node name="qt_multisense_gui" pkg="multisense_gui" type="multisense_gui.py" respawn="true">
    <param name="audio_stamped_topic" value="$(arg audio_stamped_topic)" />
  </node>


</launch>
